dev/AnimatedVisualPlayer/AnimatedVisualPlayer.cpp

```diff
@@ -371,15 +373,7 @@ void AnimatedVisualPlayer::OnLoaded(winrt::IInspectable const& /*sender*/, winrt
     // calling back into the outer.
     Background(winrt::SolidColorBrush(winrt::Colors::Transparent()));
 
-    if (m_isUnloaded)
```

Comment:
A distinction was made here between the first OnLoaded() call and subsequent calls. The comment states that UpdateContent() was not called for the first OnLoaded() call in order to not "overwrite any state that was set before the event was fired".

I checked the code and the only place where UpdateContent() was called prior to OnLoaded() was in the `OnSourcePropertyChanged()` method. I now added an extra check in UpdateContent() right at the beginning to only proceed if the AnimatedVisualPlayer control has been loaded. Thus, by the the time UpdateContent() is called in OnLoaded() we have ensured that
* all APIs of the AnimatedVisualPlayer settable via a style now have their correct values if accessed in UpdateContent() and
* there is no prior state to overwrite as UpdateContent() will only do its job once OnLoaded() has been called.

This approach looks fine to me but pinging @simeoncran to verify if I didn't accidentally miss something here and this approach should not be used.

Reply:
@simeoncran Sure.
>  It would be helpful if you could explain the root cause of the bug so that I can understand why this is the right fix. 

The issue here is that when setting the `AutoPlay` API in a style, its set value wasn't yet available when the AnimatedVisualPlayer's `Source` API was set in Markup or in code when constructing the player, for example. (So in a markup like 
```xml
<Style x:Key="NoAutoPlayAVPStyle" TargetType="AnimatedVisualPlayer">
    <Setter Property="AutoPlay" Value="False" />
</Style>

<AnimatedVisualPlayer Style="{StaticResource NoAutoPlayAVPStyle}">
    <lotties:LottieLogo/>
<AnimatedVisualPlayer>
```
even though we set `AutoPlay` to false, the lottie animation would still be played when the containing page is loaded.)

The reason for this is that when the `Source` is set, `OnSourcePropertyChanged()` is called which in turn calls `UpdateContent()`. UpdateContent() checks the value of `AutoPlay` and if it is true, then the animation will be started:
https://github.com/microsoft/microsoft-ui-xaml/blob/4d0baa4cc846de73969953b8e57772e97d8e3149/dev/AnimatedVisualPlayer/AnimatedVisualPlayer.cpp#L907

The problem is that at the time `UpdateContent()` does queries `AutoPlay`, the value returned is still `true` as that is the default value of the `AutoPlay` API and our custom style (NoAutoPlayAVPStyle) hasn't yet been applied. Thus, the animation will start playing when the page is loaded even though we told it not to in our custom style.

In other words, before we first query `AutoPlay` in UpdateContent() (for example after the source has been set) we need to make sure that a possible custom set style has been applied. This is the case when the `AnimatedVisualPlayer::OnLoaded()` method is called. As such, I changed the setup logic here so that 
* UpdateContent() will wait with doing its job and quering the `AutoPlay` API *until* the control has been loaded and
* OnLoaded() will call UpdateContent() on the *first* raised `Loaded` event now instead of just subsequent raised Loaded events to update the content and start the animation if required

This way, UpdateContent() is now run fully when the control has finished loading and can now work with the correct `AutoPlay` value.

> And can you explain the behavior of AutoPlay now with regard to how it interacts with the source being set and loading completing etc.?

The behavior isn't changed when AnimatedVisualPlayer control has *already* been loaded: Updating the source will call UpdateContent() which will start the animation automatically depending on the `AutoPlay` value. 
The only change here is when the source is set while the control has *not* been loaded. If `AutoPlay` is set to true, then the animation will only be started when the *control* has been loaded (so the Loaded event raised).

Here's a GIF, with AutoPlay set to true, showing how that would look when loading a page with a AnimatedVisualPlayer control:

```xml
<AnimatedVisualPlayer AutoPlayer="True">
    <lotties:LottieLogo/>
<AnimatedVisualPlayer>
```
![avp-autoplay](https://user-images.githubusercontent.com/1398851/91218758-d6abbf80-e719-11ea-8a60-cb639b032440.gif)

Let me know if there are still some things unclear.

Reply:
I think this looks great. Good job!
Please flip the logic and naming of m_isUnloaded to be m_isLoaded. It's not completely obvious whether the word "unloaded" means you were actually unloaded or simply not yet loaded. Previously it only meant "I have been unloaded and not yet loaded again". m_isLoaded should be unambiguous.

Reply:
@simeoncran Done.